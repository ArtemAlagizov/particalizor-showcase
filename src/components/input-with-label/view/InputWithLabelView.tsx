import React from "react";
import { Styled } from "../styled";
import { IInputWithLabelView } from "./IInputWithLabelView";
import Input from "antd/es/input";

const InputWithLabelView: React.FC<IInputWithLabelView> = ({
  onChange,
  defaultValue,
  label
}) => {
  return (
    <Styled.InputWithLabelContainer>
      <Styled.InputLabel>{label}</Styled.InputLabel>
      <Styled.InputContainer>
        <Input
          placeholder={"set background color"}
          onChange={onChange}
          defaultValue={defaultValue}
        />
      </Styled.InputContainer>
    </Styled.InputWithLabelContainer>
  );
};

export default InputWithLabelView;
