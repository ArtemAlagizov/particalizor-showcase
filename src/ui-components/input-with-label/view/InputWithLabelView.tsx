import React from "react";
import { Styled } from "../styled";
import { IInputWithLabelView } from "./IInputWithLabelView";
import { Input } from "antd";

const InputWithLabelView: React.FC<IInputWithLabelView> = ({
  onChange,
  defaultValue,
  label
}) => {
  return (
    <Styled.InputWithLabelContainer>
      <Styled.InputLabel>{label}</Styled.InputLabel>
      <Styled.InputContainer>
        <Input
          placeholder={"set background color"}
          onChange={onChange}
          defaultValue={defaultValue}
        />
      </Styled.InputContainer>
    </Styled.InputWithLabelContainer>
  );
};

export default InputWithLabelView;
