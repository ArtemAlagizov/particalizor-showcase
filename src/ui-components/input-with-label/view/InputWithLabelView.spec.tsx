import React from "react";
import { shallow } from "enzyme";
import InputWithLabelView from "./InputWithLabelView";

describe("Given a InputWithLabelView", () => {
  describe("when rendering", () => {
    it("should match snapshot", () => {
      const component = shallow(
        <InputWithLabelView onChange={jest.fn()} label={""} defaultValue={""} />
      );

      expect(component).toMatchSnapshot();
    });
  });
});
