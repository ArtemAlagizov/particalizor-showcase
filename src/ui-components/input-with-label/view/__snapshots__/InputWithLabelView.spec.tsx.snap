// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Given a InputWithLabelView when rendering should match snapshot 1`] = `
<styled.div>
  <styled.div />
  <styled.div>
    <Input
      defaultValue=""
      onChange={[MockFunction]}
      placeholder="set background color"
      type="text"
    />
  </styled.div>
</styled.div>
`;
